main.tf

```diff
@@ -227,6 +238,18 @@ data "aws_iam_policy_document" "default" {
   }
 }
 
+resource "aws_s3_bucket_notification" "bucket_notification_replication" {
+  #count  = var.notification_sns_arn != [""] && var.notification_events != [""] ? 1 : 0
+  #count  = var.replication_enabled && var.notification_enabled ? 1 : 0
+  count  = var.replication_enabled &&  var.notification_sns_arn != [""] ? 1 : 0
```

Comment:
I think you're still missing a condition from this - if you look at your other conditions you can see that `notification_events` having content is one of the evaluating conditions